elfi/post_proc.py

```diff
@@ -0,0 +1,106 @@
+import numpy as np
+from collections import OrderedDict
+import inspect
+
+import elfi.visualization as vis
+
+
+"""
+Implementations for post-processing.
+"""
+
+
+class Result(object):
+    """Container for results from ABC methods. Allows intuitive syntax for plotting etc.
+
+    Parameters
+    ----------
+    samples : OrderedDict of name: np.arrays
+    """
+    def __init__(self, samples, **kwargs):
+        self.samples = samples
+        self.n_samples = len(list(self.samples.values())[0])
+        self.n_params = len(samples)
+
+        # get name of the ABC method
+        stack10 = inspect.stack()[1][0]
+        self.method = stack10.f_locals["self"].__class__.__name__
+
+        # TODO: needed?
+        if any(map(lambda k: k in self.__dir__(), kwargs.keys())):
+            raise KeyError("Conflicting key")
+
+        self.__dict__.update(kwargs)
+
+    def __str__(self):
+        return self.summarize(stdout=False)
+
+    def __repr__(self):
+        return self.__str__()
+
+    def summarize(self, stdout=True):
```

Comment:
Why do we carry a `stdout=True` flag around instead of more generic `output_stream`, for example? Or better, why don't we just always return a string and let the caller do the printing to stream (one function, one responsibility)?